File: ontrack-job/src/main/java/net/nemerosa/ontrack/job/JobRegistration.java
package net.nemerosa.ontrack.job;

import lombok.Data;
import lombok.experimental.Wither;

/**
 * Association of a job and a schedule for a registration at startup.
 */
@Data
public class JobRegistration {

    private final Job job;
    @Wither
    private final Schedule schedule;

    public static JobRegistration of(Job job) {
        return new JobRegistration(job, Schedule.NONE);
    }

    public JobRegistration everyMinutes(long minutes) {
        return withSchedule(Schedule.everyMinutes(minutes));
    }
}

File: ontrack-job/src/main/java/net/nemerosa/ontrack/job/JobRegistration.kt
package net.nemerosa.ontrack.job

/**
 * Association of a job and a schedule for a registration at startup.
 */
data class JobRegistration(
        val job: Job,
        val schedule: Schedule
) {

    fun withSchedule(schedule: Schedule) = JobRegistration(job, schedule)

    fun everyMinutes(minutes: Long): JobRegistration {
        return withSchedule(Schedule.everyMinutes(minutes))
    }

    companion object {
        @JvmStatic
        fun of(job: Job): JobRegistration {
            return JobRegistration(job, Schedule.NONE)
        }
    }
}
